### DISCUSSION TITLE
After executing retention policies & GC, how can I undo the retention jobs

### DISCUSSION THREAD

**Discussion by mfernandezfunes** [goharbor/harbor#19579] [Q&A]
I setted a retention rule, I execute the GC to start working. In the process I noticed that a lot of images started being deleted that weren't supposed to, so I stopped the GC. Is there a way to unmark the reamining artifacts from being deleted..?
I saw that blobs from the S3 bucket can be undeleted, but what else can I check ?
Is there any docuemntation about how GC works? Thank u

**Comment by AllForNothing** [comment]
@mfernandezfunes You refer the related section in our documentation https://goharbor.io/docs/2.9.0/administration/garbage-collection & https://goharbor.io/docs/2.9.0/working-with-projects/working-with-images/create-tag-retention-rules & https://goharbor.io/docs/2.9.0/working-with-projects/working-with-images/create-tag-immutability-rules

You can set the `tag-immutable-rule` to protect the images from being deleted

And there is no way to undo the deletion, so recommend you run GC with `dry-run` mode before first

**Reply by mfernandezfunes** [reply]
The problem is that after executing retention rules in dry-run mode, and checking Policy retention logs it shows that the retention was well applied, but after the execution of the GC, some blobs were deleted not applying to the rules

**Reply by AllForNothing** [reply]
@mfernandezfunes Do the deleted images have tags?

**Reply by mfernandezfunes** [reply]
Yes they did have

**Reply by AllForNothing** [reply]
@mfernandezfunes By design, GC will not delete any referenced images.  If possible, can you move this to the issues tab, and provide the necessary logs?
